### DISCUSSION TITLE
Error - Deployment to Shiny

### DISCUSSION THREAD

**Discussion by DGUM27** [teamalgoritma/community#272] [Q&A] [answered]
Dear Bapak/Ibu, 

Saya mengalami kendala dalam deploy ke shiny.io dengan keterangan 

"**An error has occurred!
An error has occurred. Check your logs or contact the app author for clarification.**"

Please your help to fix this issue. 

Thank you. 

Attachment:

![A](https://user-images.githubusercontent.com/107378736/185292741-86ca09c4-1478-4f68-a5d2-694de252d676.jpeg)

**Comment by anugrah18003** [comment, marked as answer]
Dear @DGUM27 ,

Untuk mengetahui informasi error yang terjadi, silahkan untuk melakukan pengecekan pada log shinyapps yang dideploy pada akun shinyapps.io Anda. Boleh dicek pada bagian log app terbawah jika terdapat pesan error, maka Anda perlu memperbaiki error yang terjadi tersebut.

Selain itu izin memastikan apakah saat menjalankan dashboard secara lokal mengalami error yang sama? atau hal ini hanya terjadi saat melakukan deployment di shinyapps?

**Reply by DGUM27** [reply]
Terima kasih atas response-nya @anugrah18003 

Boleh saya tahu di bagian mana di tampilan shinyapps.io untuk menegecek log tersebut? *saya lampirkan screen capture saya di bawah. Untuk dashboard dapat berjalan secara lokal sebagaimana juga saya lampirkan.

Please your help @anugrah18003 

Thanks in advance. 

Attachment:

1. Tampilan shinyapps.io saya

![B](https://user-images.githubusercontent.com/107378736/185312260-5b90077f-49a5-437d-9b82-4ec522151c22.png)

2. Tampilan dashboard yang saya running di lokal saya
 
![C](https://user-images.githubusercontent.com/107378736/185312770-74db1cc7-df8b-46f2-b603-02040672a956.png)

**Reply by DGUM27** [reply]
@anugrah18003 

Terlampir untuk logs di shiny app saya


![D](https://user-images.githubusercontent.com/107378736/185320688-3538f1ba-423b-4d0e-9899-a78ba69094fd.png)

**Reply by anugrah18003** [reply]
@DGUM27 

Baik terimakasih, jika saya lihat dari logs di shiny app terdapat error message yaitu could not find function "dashboardPage", saya menduga ada packages yang belum terinstall.

Boleh saya lihat juga global, ui dan servernya? agar saya bisa melihat codingannya~

**Reply by DGUM27** [reply]
Thank you @anugrah18003 ,

attached untuk ui, server, dan global saya. 

[DG.zip](https://github.com/teamalgoritma/community/files/9371747/DG.zip)

please your help

**Reply by anugrah18003** [reply]
Dear @DGUM27 

Setelah saya perhatikan kembali pada file global yang bapak gunakan belum memasukan library yang akan digunakan. Silahkan untuk tuliskan library-library yang digunakan pada dashboard ini. Tidak adanya import library ini dapat menyebabkan error "function not found" karena biasanya function tersebut berasal dari packages tertentu.
Note: Pada saat upload ke web, library-library tersebut akan diimport oleh website shinyapps sehingga perlu dimasukan kedalam syntax, berbeda dengan saat dijalankan dilocal/device sendiri dimana library-library tersebut sudah terdownload dan pernah digunakan.
![Capture](https://user-images.githubusercontent.com/70314445/185530277-2dc104b1-0dd5-4047-b73e-13a86ab7cbd6.PNG)

Selain itu silahkan untuk ubah format filenya dari global.Rmd menjadi global.R agar sama dengan ui.R dan server.R

Silahkan dicoba kembali jika sudah melakukan hal-hal tersebut. Semoga membantu~

**Reply by DGUM27** [reply]
Terima kasih banyak @anugrah18003  saya sudah dapat men-deploy-nya.

Saya juga mengalami kendala ketika menggunakan sliderinput, dimana expected result-nya adalah "tahun dengan bilangan tanpa coma" namun masih terdapat tanda "coma di dalam sliderinput tersebut"

Mohon advice-nya @anugrah18003  

Terima kasih 

attachment:
![E](https://user-images.githubusercontent.com/107378736/185562523-2ac70a87-3e8c-4d7d-9e11-9da400eb22d5.png)

![F](https://user-images.githubusercontent.com/107378736/185562586-440e119c-a3ed-452e-8f84-f55ca1e6069b.png)

**Reply by anugrah18003** [reply]
Dear @DGUM27 

Sebelumnya boleh saya lihat syntax dibagian server dan ui untuk sliderinput-nya?

**Reply by DGUM27** [reply]
Dear @anugrah18003 

Sudah solved.

Thank you sudah membantu saya.
